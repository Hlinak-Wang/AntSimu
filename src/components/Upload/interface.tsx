export interface IUploadFile {
  uid: number,
  size: number,
  name: string,
  status?: 'success' | 'onload' | 'error',
  percent?: number,
  file?: File,
  response?: any,
  error?: any,
}

export interface IUploadRef {
  focus: () => void;
  blur: () => void;
  click: () => void;
  upload: (files: FileList) => void;
}

export interface IUploadCore {
  /**上传了的文件 */
  fileList?: IUploadFile[];
  /**接受上传的文件类型 */
  accept?: string[];
  /**上传请求的 http method */
  method?: 'post' | 'delete' | 'get' | 'put';
  /**是否可多选 */
  multiple?: boolean;
  /**上传的文件名 */
  filename?: string | ((file: File) => string);
  /**上传的地址 */
  action: string | ((file: File) => PromiseLike<string>);
  /**设置上传的请求头部 */
  headers?: object;
  /**上传请求时是否携带 cookie */
  withCredentials?: boolean;
  /**	
   * 上传文件之前的钩子，参数为上传的文件，若返回 false 则停止上传。
   * 支持返回一个 Promise 对象，Promise 对象 reject 时则停止上传，
   * resolve 时开始上传（ resolve 传入 File 或 Blob 对象则上传 resolve 传入对象） 
   */
  beforeUpload?: (file: File) => boolean | PromiseLike<File>;
  /** 上传时的回调 */
  onProgress?: (percent: number, file: File) => void;
  /** */
  onDownLoad?: (file: File) => void;
  /** 上传成功时的回调 */
  onSuccess?: (response: object, file: File) => void;
  /**上传失败时的回调 */
  onError?: (error: Error, file: File) => void;
  /**	上传文件改变时的状态 */
  onChange?: (files: IUploadFile[]) => void;
}

export interface IUploadProps extends IUploadCore {
  /**默认已经上传的文件列表 */
  defaultFileList?: IUploadFile[];
  /**是否支持上传文件夹 */
  directory?: boolean;  
  /**是否禁用 */
  disabled?: boolean;
  /**上传列表的内建样式 */
  listType?: 'text' | 'picture';
  /**上传所需额外参数或返回上传额外参数的方法 */
  data?: object | ((file: File) => object);
  /**
   * 是否使用拖拽
   *  @default false
   */
  dragEnable?: boolean;
  /** */
  onDownLoad?: (file: File) => void;
  /**点击移除文件时的回调，返回值为 false 时不移除。
   * 支持返回一个 Promise 对象，Promise 对象 resolve(false) 或 reject 时不移除。
   */
  onRemove?: (file: IUploadFile) => void;
  /**点击文件或图片时的回调 */
  onPreview?: (file: IUploadFile) => void;
}