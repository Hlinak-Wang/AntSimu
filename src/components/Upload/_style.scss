$upload-file-status: (
  'success': $green,
  'onload': $primary,
  'error': $red,
);

.upload-file-group {
  padding: 0;
  margin: 0;
}

.upload-file {
  list-style: none;
  padding: 0.25rem 0.5rem;
  border-radius: 2px;
  transition: background 0.3s ease-in-out;
  position: relative;

  .upload-file-icon {
    margin-right: 5px;
  }

  .upload-file-action {
    text-align: center;
    line-height: inherit;
    color: $gray-7;
    display: none;
    width: 1rem;
    height: 1rem;
    float: right;
    cursor: pointer;
    &:hover {
      color: $black;
    }
  }

  .upload-file-status {
    text-align: center;
    float: right;
    display: block;
    width: 1rem;
    height: 1rem;
    cursor: pointer;
    line-height: inherit;
  }

  @each $key, $value in $upload-file-status {
    &.upload-file-#{$key} {
      @if $key == 'error' {
        color: $value;
      } @else {
        .upload-file-status {
          color: $value;
        }
      }
    }
  }

  &:hover {
    background: $gray-3;
    .upload-file-action {
      display: block;
    }
    .upload-file-status {
      display: none;
    }
  }
}

.hlinak-upload-dragger-container {
  
  width: 250px;
  height: 200px;
  border: 1px dashed $gray-4;
  border-radius: 2px;
  background: $gray-1;
  text-align: center;
  cursor: pointer;
  transition: border-color 0.3s ease-in;
  &.is-dragOver {
    border: 1px dashed $primary;
    background: lighten($primary, 30%)
  }
  .hlinak-upload-dragger-content {
    padding: 1rem 0;
    display: table;
  }
}

.hlinak-upload-trigger {
  display: inline-block;
  //pointer-events: none;
}