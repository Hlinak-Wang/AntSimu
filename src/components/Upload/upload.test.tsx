import React from 'react';
import { render, fireEvent, RenderResult, wait, queryByTestId, cleanup } from '@testing-library/react';
import Upload from './upload';
import { IUploadProps } from './interface';
import axios from 'axios';
import { act } from 'react-dom/test-utils';

//jest.mock('axios');
/* jest.mock('rc-tween-one', () => {
  return {
    TweenOneGroup: (props: any) => <ul>{props.children}</ul>
  }
}) */

//const jestAxios = axios as jest.Mocked<typeof axios>;
let wrapper: RenderResult, fileInput: HTMLInputElement, uploadArea: HTMLDivElement;

async function checkFileUpload(
  wrapper: RenderResult, 
  testProps: Pick<IUploadProps, "onSuccess" | "onProgress" | "onChange">, 
  fileCalled: File, 
  expectStatus: {onload: number, success?: number, error?: number}
  ) {
  expect(testProps.onChange).toBeCalled();
  await wait(() => {
    expect(wrapper.getAllByTestId('spin').length).toBe(expectStatus.onload);
  })
  
  await wait(() => {
    expect(wrapper.getAllByTestId('check-circle').length).toBe(expectStatus.success);
    expect(wrapper.queryByText(fileCalled.name)).toBeInTheDocument();
  })

  expect(testProps.onProgress).toBeCalled();
  //expect(queryByText()) 
  expect(testProps.onSuccess).toBeCalledWith({'test': "ok"}, fileCalled)
}

describe("test for action", () => {
  const testProps: Partial<IUploadProps> = {
    method: 'post',
    onSuccess: jest.fn(),
    onError: jest.fn(),
    onChange: jest.fn(),
    onProgress: jest.fn(),
  }
  it("action is string", async () => {
    
    wrapper = render(<Upload {...testProps} action="http://www.mocky.io/v2/5ec939dc2f0000354adb719a">upload</Upload>);
    fileInput = wrapper.container.querySelector("input") as HTMLInputElement;
    expect(wrapper).toMatchSnapshot();

    expect(wrapper).toMatchSnapshot();
    const testFile = new File(Array(1), "test.txt");
    act(() => {
      fireEvent.change(fileInput, { target: { files: [testFile] }});
    })
    
    await checkFileUpload(wrapper, testProps, testFile, {onload: 1, success: 1});
  })
})

describe('basic test', () => {
  const testProps: IUploadProps = {
    action: "http://www.mocky.io/v2/5ec939dc2f0000354adb719a", // return { "test": "ok" }
    method: 'post',
    onSuccess: jest.fn(),
    onError: jest.fn(),
    onChange: jest.fn(),
    onProgress: jest.fn(),
  }

  it("test post multiple file seperately and at the same time", async () => {
    wrapper = render(<Upload {...testProps} multiple>upload</Upload>);
    fileInput = wrapper.container.querySelector('input') as HTMLInputElement;

    expect(wrapper).toMatchSnapshot();
    const file1 = new File(Array(1), "file1.txt");
    act(() => {
      fireEvent.change(fileInput, { target: { files: [file1] }});
    })

    expect(testProps.onChange).toBeCalled();
    await checkFileUpload(wrapper, testProps, file1, {onload: 1, success: 1});

    const file2 = new File(Array(2), "file2.txt")
    act(() => {
      fireEvent.change(fileInput, { target: { files: [file2] }});
    })
    await checkFileUpload(wrapper, testProps, file2, {onload: 1, success: 2});
  })

  it('test beforeUpload which return boolean', async () => {
    const wrapper = render(
      <Upload 
        {...testProps} 
        beforeUpload={(file) => {
          if (file.size > 10) {
            return false;
          } else {
            return true
          }
        }}
      >
        upload
      </Upload>
    );
    const fileInput = wrapper.container.querySelector('input') as HTMLInputElement;
    expect(wrapper).toMatchSnapshot();
    
    const smallFile = new File(Array(1), "small.txt")
    act(() => {
      fireEvent.change(fileInput, { target: { files: [smallFile] }});
    })

    expect(testProps.onChange).toBeCalled();

    await wait(() => {
      expect(wrapper.getByTestId('spin')).toBeInTheDocument();
    })
    
    await wait(() => {
      expect(wrapper.getByTestId('check-circle')).toBeInTheDocument();
      expect(wrapper.queryByText('small.txt')).toBeInTheDocument();
    })

    const largeFile = new File(Array(10), "large.txt");
    act(() => {
      fireEvent.change(fileInput, { target: {files: [largeFile]}});
    })
    
    //expect(testProps.onChange).not.toBeCalled();
  })

  it("test beforeUpload return promise", async () => {
    const wrapper = render(
      <Upload 
        {...testProps} 
        beforeUpload={(file) => {
          const transformedFile = new File([file], "new.txt");
          return Promise.resolve(transformedFile)
        }}
      >
        upload
      </Upload>
    );
    const fileInput = wrapper.container.querySelector('input') as HTMLInputElement;
    expect(wrapper).toMatchSnapshot();

    const originFile = new File(Array(1), "origin.txt")
    act(() => {
      fireEvent.change(fileInput, { target: { files: [originFile] }});
    })

    expect(testProps.onChange).toBeCalled();

    await wait(() => {
      expect(wrapper.getByTestId('spin')).toBeInTheDocument();
    })
    
    await wait(() => {
      expect(wrapper.getByTestId('check-circle')).toBeInTheDocument();
      expect(wrapper.queryByText('new.txt')).toBeInTheDocument();
    })
  })
})

