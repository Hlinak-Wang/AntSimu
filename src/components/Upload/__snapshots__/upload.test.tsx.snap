// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic test should post single file 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="hlinak-upload-component"
      >
        <div
          class="hlinak-upload-trigger"
        >
          upload
        </div>
        <input
          style="display: none;"
          type="file"
        />
        <ul
          class="upload-file-group"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="hlinak-upload-component"
    >
      <div
        class="hlinak-upload-trigger"
      >
        upload
      </div>
      <input
        style="display: none;"
        type="file"
      />
      <ul
        class="upload-file-group"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`basic test test beforeUpload 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="hlinak-upload-component"
      >
        <div
          class="hlinak-upload-trigger"
        >
          upload
        </div>
        <input
          style="display: none;"
          type="file"
        />
        <ul
          class="upload-file-group"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="hlinak-upload-component"
    >
      <div
        class="hlinak-upload-trigger"
      >
        upload
      </div>
      <input
        style="display: none;"
        type="file"
      />
      <ul
        class="upload-file-group"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`basic test test beforeUpload return promise 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="hlinak-upload-component"
      >
        <div
          class="hlinak-upload-trigger"
        >
          upload
        </div>
        <input
          style="display: none;"
          type="file"
        />
        <ul
          class="upload-file-group"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="hlinak-upload-component"
    >
      <div
        class="hlinak-upload-trigger"
      >
        upload
      </div>
      <input
        style="display: none;"
        type="file"
      />
      <ul
        class="upload-file-group"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`basic test test beforeUpload w 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="hlinak-upload-component"
      >
        <div
          class="hlinak-upload-trigger"
        >
          upload
        </div>
        <input
          style="display: none;"
          type="file"
        />
        <ul
          class="upload-file-group"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="hlinak-upload-component"
    >
      <div
        class="hlinak-upload-trigger"
      >
        upload
      </div>
      <input
        style="display: none;"
        type="file"
      />
      <ul
        class="upload-file-group"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`basic test test beforeUpload which return boolean 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="hlinak-upload-component"
      >
        <div
          class="hlinak-upload-trigger"
        >
          upload
        </div>
        <input
          style="display: none;"
          type="file"
        />
        <ul
          class="upload-file-group"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="hlinak-upload-component"
    >
      <div
        class="hlinak-upload-trigger"
      >
        upload
      </div>
      <input
        style="display: none;"
        type="file"
      />
      <ul
        class="upload-file-group"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`basic test test beforeUpload which return file 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="hlinak-upload-component"
      >
        <div
          class="hlinak-upload-trigger"
        >
          upload
        </div>
        <input
          style="display: none;"
          type="file"
        />
        <ul
          class="upload-file-group"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="hlinak-upload-component"
    >
      <div
        class="hlinak-upload-trigger"
      >
        upload
      </div>
      <input
        style="display: none;"
        type="file"
      />
      <ul
        class="upload-file-group"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`basic test test post multiple file seperately and at the same time 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="hlinak-upload-component"
      >
        <div
          class="hlinak-upload-trigger"
        >
          upload
        </div>
        <input
          multiple=""
          style="display: none;"
          type="file"
        />
        <ul
          class="upload-file-group"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="hlinak-upload-component"
    >
      <div
        class="hlinak-upload-trigger"
      >
        upload
      </div>
      <input
        multiple=""
        style="display: none;"
        type="file"
      />
      <ul
        class="upload-file-group"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`basic test test post single file 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="hlinak-upload-component"
      >
        <div
          class="hlinak-upload-trigger"
        >
          upload
        </div>
        <input
          style="display: none;"
          type="file"
        />
        <ul
          class="upload-file-group"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="hlinak-upload-component"
    >
      <div
        class="hlinak-upload-trigger"
      >
        upload
      </div>
      <input
        style="display: none;"
        type="file"
      />
      <ul
        class="upload-file-group"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`test for action action is string 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="hlinak-upload-component"
      >
        <div
          class="hlinak-upload-trigger"
        >
          upload
        </div>
        <input
          style="display: none;"
          type="file"
        />
        <ul
          class="upload-file-group"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="hlinak-upload-component"
    >
      <div
        class="hlinak-upload-trigger"
      >
        upload
      </div>
      <input
        style="display: none;"
        type="file"
      />
      <ul
        class="upload-file-group"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`test for action action is string 2`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="hlinak-upload-component"
      >
        <div
          class="hlinak-upload-trigger"
        >
          upload
        </div>
        <input
          style="display: none;"
          type="file"
        />
        <ul
          class="upload-file-group"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="hlinak-upload-component"
    >
      <div
        class="hlinak-upload-trigger"
      >
        upload
      </div>
      <input
        style="display: none;"
        type="file"
      />
      <ul
        class="upload-file-group"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
